[8.16] Updates docs for and related to the `excludedDataTiersForRuleExecution` advanced setting (#5962)
elastic/security-docs: docs/detections/detections-exclude-cold-frozen-data-tiers.asciidoc

[[exclude-cold-frozen-data-individual-rules]]

> == Exclude cold and frozen data from rule executions

Reviewer: I think the title should be updated to make it more clear and explicit that this workflow applies to **SINGLE** rules, not **ALL** rules.

Author: @peluja1012 how's this:

```suggestion
== Exclude cold and frozen data from rule executions for individual rules
```

Author: @marshallmain @peluja1012 how's this:

```suggestion
== Exclude cold and frozen data from individual rules
```

Reviewer: Thanks! look good to me.

:frontmatter-description: Configure a rule to ignore cold and frozen data during execution. 
:frontmatter-tags-products: [security]
:frontmatter-tags-content-type: [how-to]
:frontmatter-tags-user-goals: [manage]

> Your rule might perform slower if it queries data from cold and frozen {ref}/data-tiers.html[data tiers]. To help Elasticsearch efficiently exclude cold and frozen data, apply a Query DSL filter that ignores cold and frozen documents when your rule executes. You can add the filter when creating a new rule or updating an existing one.

Author: Feedback from Mike P.

```suggestion
Your rule might perform slower or fail if it queries data from cold and frozen {ref}/data-tiers.html[data tiers]. To help Elasticsearch efficiently exclude cold and frozen data, apply a Query DSL filter that ignores cold and frozen documents when your rule executes. You can add the filter when creating a new rule or updating an existing one.
```

> Rules that query cold and frozen {ref}/data-tiers.html[data tiers] might perform more slowly. To exclude query results from cold and frozen tiers, add a Query DSL filter that ignores cold and frozen documents when executing. This can help Elasticsearch exclude cold and frozen data more efficiently. You can add the filter when creating a new rule or updating an existing one.

Reviewer: Could we make it more clear and explicit that this workflow applies to **SINGLE** rules, not **ALL** rules?

Author: Good call, @peluja1012. How's this:

```suggestion
Your rule might perform slower if it queries data from cold and frozen {ref}/data-tiers.html[data tiers]. To help Elasticsearch efficiently exclude cold and frozen data, apply a Query DSL filter that ignores cold and frozen documents when your rule executes. You can add the filter when creating a new rule or updating an existing one.
```

cc: @yctercero @marshallmain

Author: Verified this change with @peluja1012 over slack.

> TIP: To ensure that _all_ rules in a {kib} space exclude cold and frozen documents when executing, configure the `excludedDataTiersForRuleExecution` <<exclude-cold-frozen-data-rule-executions,advanced setting>>.

Author: ```suggestion
TIP: To ensure that rules in a {kib} space exclude cold and frozen documents when executing, configure the `excludedDataTiersForRuleExecution` <<exclude-cold-frozen-data-rule-executions,advanced setting>>.
```

[IMPORTANT]
====

> * This method is not supported for {esql} and {ml} rules.

Reviewer 2: The setting is supported for ES|QL rules and may help performance when frozen shards are available, but it won't prevent failures when frozen shards are unavailable. The setting has no effect on ML rules.

Author: > The setting is supported for ES|QL rules and may help performance when frozen shards are available,

@marshallmain to clarify, are you saying that using a **DSL filter** to exclude query results from cold and frozen tiers does work for ES|QL rules? I know the filter works similarly to the `excludedDataTiersForRuleExecution` setting, but I still want to clarify in case my understanding isn't correct.

Reviewer 2: Sorry I missed that this was about adding the filter manually rather than the advanced setting - it's correct as is 👍

> * Indicator match, event correlation, and {esql} rules might fail if a frozen or cold shard that matches the rule's specified index pattern is unavailable during rule executions. The best path forward continues to be modifying the index patterns to only use hot tier data.

Author: Feedback from Mike P.

```suggestion
* Even when applying this filter, indicator match and event correlation rules may still fail if a frozen or cold shard that matches the rule's specified index pattern is unavailable during rule executions. If failures occur, we recommend modifying the rule's index patterns to only match indices containing hot tier data.

```

> * Some rule types (specifically the indicator match, event correlation, and {esql} rule types) might fail if a frozen or cold shard that matches the rule's specified index pattern is unavailable during rule executions. The best path forward continues to be modifying the index patterns to only use hot tier data.

Reviewer 3: ```suggestion
* Some rule types (specifically the indicator match, event correlation, and {esql} rule types) might fail if a frozen or cold shard that matches the rule's specified index pattern is unavailable during rule executions. The best path forward continues to be modifying the index patterns to only match data in the tiers you would like to target.
```

Reviewer 2: ```suggestion
* Indicator match, event correlation, and {esql} rules might fail if a frozen or cold shard that matches the rule's specified index pattern is unavailable during rule executions. The best path forward continues to be modifying the index patterns to only use hot tier data.
```

[[exclude-cold-frozen-data-individual-rules]]
== Exclude cold and frozen data from individual rules

:frontmatter-description: Configure a rule to ignore cold and frozen data during execution. 
:frontmatter-tags-products: [security]
:frontmatter-tags-content-type: [how-to]
:frontmatter-tags-user-goals: [manage]

Your rule might perform slower or fail if it queries data from cold and frozen {ref}/data-tiers.html[data tiers]. To help Elasticsearch exclude cold and frozen data more efficiently, apply a Query DSL filter that ignores cold and frozen documents when your rule executes. You can add the filter when creating a new rule or updating an existing one.

> TIP: To ensure that the rules in a {kib} space exclude cold and frozen documents when executing, configure the `excludedDataTiersForRuleExecution` <<exclude-cold-frozen-data-rule-executions,advanced setting>>.

Author: ```suggestion
TIP: To ensure that the rules in your {kib} space exclude cold and frozen documents when executing, configure the `excludedDataTiersForRuleExecution` <<exclude-cold-frozen-data-rule-executions,advanced setting>>.
```

[IMPORTANT]
====

* This method is not supported for {esql} and {ml} rules.

> * Even when applying this filter, indicator match and event correlation rules may still fail if a frozen or cold shard that matches the rule's specified index pattern is unavailable during rule executions. If failures occur, we recommend modifying the rule's index patterns to only match indices containing hot tier data.

Author: @yctercero @marshallmain need your input on this ^ please and thank you!

> * Even when applying this filter, indicator match and event correlation rules may still fail if a frozen or cold shard that matches the rule's specified index pattern is unavailable during rule executions. If failures occur, we recommend modifying the rule's index patterns to only match indices containing hot tier data.

Reviewer 3: Let's keep it to `hot` only at the moment. We'll need to conduct more research around `warm` tiers, but we know for sure that `hot` will work.
